### ISSUE TITLE
Allow sprig expressions in WorkflowEventBinding submit metadata

### ISSUE THREAD

**Issue by svmaris** [argoproj/argo-workflows#5351]
# Summary

#4640 added support for expressions in the name of webhook-submitted workflows. Please allow sprig functions as well.

# Use Cases

We're in the same boat as @michaeljohnalbers . We're using Bitbucket Server Webhooks to trigger generic build pipelines in Argo Workflows. Multiple services (repos) are handled by the same WorkflowTemplate. When submitting new Workflows, we'd like to see the actual service being built as part of the name of the workflow.

It seems that only 'simple' expressions are allowed right now, eg. things coming directly from the event payload or metadata. We're having a hard time getting the right naming scheme, because it has to be unique *and* we're limited to the Kubernetes DNS naming scheme (so it can't be too long).

We have no control over the information coming from Bitbucket Server. Ideally, we'd like to add the commit hash, but the payload only contains the full SHA256 hash, which is too long when combined with the service name. Also, there's no unix timestamp in either the headers or the payload we can use.

If we could do something like this:

```yaml
spec:
  event:
    selector: ...
  submit:
    metadata:
      name: '{{payload.repository.slug}} + "-" + {{payload.changes[0].toHash | trunc 8}}'

```

it would be absolutely awesome.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
Would you like to submit a PR? You'd need to make changes:

https://github.com/argoproj/argo-workflows/blob/80b5ab9b8e35b4dba71396062abe32918cd76ddd/server/event/dispatch/operation.go#L189

Similar to:

https://github.com/argoproj/argo-workflows/blob/master/util/template/template.go#L34

**Comment by svmaris** [comment]
@alexec After spending half a day on figuring out how to get `template.Replace()` working, I finally realised I just needed to get the extra helper functions in the `env` for `expr` 🤦 . Fix is in #5373.
